src/adapter/generic/adapter_core.py

```diff
@@ -62,7 +78,7 @@ def on_close(self):
         """ Connection with AMP has been closed. Try to reconnect. """
         self.state = State.DISCONNECTED
         self.handler.stop()
```

Comment:
Do we need to empty the queues on error or close to prevent any trailing messages being sent to AMP?

Reply:
Good one! I will update that.

Reply:
... on the other hand, after the `self.handler.stop()`, the adapter will call `self.start()`, which will also clear the queue of messages.